src/transformers/trainer.py

```diff
@@ -244,6 +244,8 @@ def __init__(
             self.scaler = torch.cuda.amp.GradScaler()
 
     def _remove_unused_columns(self, dataset: "nlp.Dataset", description: Optional[str] = None):
+        if not self.ars.remove_unused_columns:
```

Comment:
Shouldn't that be `self.args` ? Feel like it'd be nice to have a test there if that kind of typo can sneak through the CI.

Reply:
Trainer is barely tested. Adding tests is on my TODO. Will make sure to add one with NLP.